### ISSUE TITLE
Baggy cross operators

### ISSUE THREAD

**Issue by nwellnhof** [rakudo/rakudo#1726]
Some baggy cross operators seem to produce wrong results. Tested version: Rakudo version 2018.03 built on MoarVM version 2018.03 implementing Perl 6.c

### Baggy multiplication

```
say <a 1> ⊍ <a 2>; # Bag(a)
say (<a 1>,) X⊍ (<a 2>,); # (Bag(1, 2, a(2))) seems to use baggy addition
say cross((<a 1>,), (<a 2>,), :with(&infix:<⊍>)); # Also wrong
say cross((<a 1>,), (<a 2>,), :with(* ⊍ *)); # Works
```

### Union of bags

```
say <a 1>.Bag ∪ <a 2>.Bag; # Bag(1, 2, a)
say (<a 1>.Bag,) X∪ (<a 2>.Bag,); # (set(Bag(1, a) Bag(2, a)))
say cross((<a 1>.Bag,), (<a 2>.Bag,), :with(&infix:<∪>)); # Also wrong
say cross((<a 1>.Bag,), (<a 2>.Bag,), :with(* ∪ *)); # Works
```

**Comment by dogbert17** [comment]
Fixed by Lizmat++ with 338c7439d2ebd

**Comment by ab5tract** [comment]
Tests added with [7749803](https://github.com/Raku/roast/commit/7749803e24b0417326fbf51ac7e4509df2f4421b) in roast.
